### PR TITLE
Sprint 8 permissions

### PR DESCRIPTION
**Pull request by AbdulSir**
All features are protected by user roles. the available roles are only Administration, Management, Sales Rep, Reception

### CODE REVIEW

**Review by purplesmurf1998 — APPROVED** [review]
Fairly straight forward changes. Looks good

File: backend/models/User.js [hunk 1]
```diff
@@ -29,37 +29,34 @@ const UserSchema = new mongoose.Schema({
...
-      'Sell Vehicles',
-      'Delete Vehicles',
-      'Add Vehicle Property',
-      'Edit Vehicle Properties',
-      'View Vehicle Properties',
-      'Delete Vehicle Properties'
-    ],
-    required: 'User must have a list of permissions, even if that list is empty.'
-  },
+  // permissions: {
```
Comment (TZnoj) [inline backend/models/User.js:36]: I'm approving this for now but the commented code needs to be removed after testing

**Review by TZnoj — APPROVED** [review]
Approved as per my comment

File: backend/models/User.js (hunk 1 above)
Comment (AbdulSir) [reply backend/models/User.js:36]: Sure, that's the plan. We'd need to perform a end to end testing then we can remove this commented code.

File: frontend/src/modules/users/StaffAccountAdd.vue [hunk 2]
```diff
@@ -252,7 +248,7 @@ export default {
         email: this.email,
         role: this.role,
         dealership: this.$route.params.dealershipId,
-        permissions: this.permissions,
+        // permissions: this.permissions,
```
Comment (AdrienKamran) [inline frontend/src/modules/users/StaffAccountAdd.vue:251]: Once the testing phase is done, we'll have to remove this code.

**Review by AdrienKamran — APPROVED** [review]
As mentioned in the individual files, once the testing phase is done, we'll have to remove the commented-out code for the new permissions system. Otherwise, this refactoring is an elegant solution to the old permissions system's biggest flaws.

File: backend/models/User.js [hunk 3]
```diff
@@ -29,37 +29,34 @@ const UserSchema = new mongoose.Schema({
...
+  //     'Edit Vehicle Locations',
+  //     'Sell Vehicles',
+  //     'Delete Vehicles',
+  //     'Add Vehicle Property',
+  //     'Edit Vehicle Properties',
+  //     'View Vehicle Properties',
+  //     'Delete Vehicle Properties'
+  //   ],
+  //   required: 'User must have a list of permissions, even if that list is empty.'
+  // },
```
[the remark this answers is not in the document]
Comment (AbdulSir) [reply backend/models/User.js:59]: Yes, that is the plan.

**Pull request merged**